src/backend/dx11/src/device.rs

```diff
@@ -1319,9 +1394,22 @@ impl hal::Device<Backend> for Device {
                     pso::Descriptor::Buffer(buffer, ref _range) => {
                         match ty {
                             pso::DescriptorType::UniformBuffer => {
+                                if buffer.ty == MemoryHeapFlags::HOST_COHERENT {
+                                    let old_buffer = unsafe { (*handle).0 } as *mut _;
+
+                                    write.set.flush_coherent(old_buffer, buffer);
```

Comment:
do we have to flush at `write_descriptor_sets`?

Reply:
It only marks the descriptor set as having descriptor(s) referring to coherent buffers. On `bind_graphics_descriptor_sets` they are then added to the command buffer.